src/ol/format/IIIFInfo.js

```diff
@@ -0,0 +1,368 @@
+ * indicates support for that quality.
+ */
+
+/**
+ * @typedef {Object} SupportedFeatures
+ * @property {Array<string>} [supports] Supported IIIF image size and region
+ * calculation features.
+ * @property {Array<string>} [formats] Supported image formats.
+ * @property {Array<string>} [qualities] Supported IIIF image qualities.
+ */
+
+/**
+ * Supported image formats, qualities and supported region / size calculation features
+ * for different image API versions and compliance levels
+ * @const
+ * @type {Object<string, Object<string, SupportedFeatures>}
+ */
+const IIIF_PROFILE_VALUES = {
+  version1: {
+    level0: {
+      supports: [],
+      formats: [],
+      qualities: ['native']
+    },
+    level1: {
+      supports: ['regionByPx', 'sizeByW', 'sizeByH', 'sizeByPct'],
+      formats: ['jpg'],
+      qualities: ['native']
+    },
+    level2: {
+      supports: ['regionByPx', 'regionByPct', 'sizeByW', 'sizeByH', 'sizeByPct',
+        'sizeByConfinedWh', 'sizeByWh'],
+      formats: ['jpg', 'png'],
+      qualities: ['native', 'color', 'grey', 'bitonal']
+    }
+  },
+  version2: {
+    level0: {
+      supports: [],
+      formats: ['jpg'],
+      qualities: ['default']
+    },
+    level1: {
+      supports: ['regionByPx', 'sizeByW', 'sizeByH', 'sizeByPct'],
+      formats: ['jpg'],
+      qualities: ['default']
+    },
+    level2: {
+      supports: ['regionByPx', 'regionByPct', 'sizeByW', 'sizeByH', 'sizeByPct',
+        'sizeByConfinedWh', 'sizeByDistortedWh', 'sizeByWh'],
+      formats: ['jpg', 'png'],
+      qualities: ['default', 'bitonal']
+    }
+  },
+  version3: {
+    level0: {
+      supports: [],
+      formats: ['jpg'],
+      qualities: ['default']
+    },
+    level1: {
+      supports: ['regionByPx', 'regionSquare', 'sizeByW', 'sizeByH'],
+      formats: ['jpg'],
+      qualities: ['default']
+    },
+    level2: {
+      supports: ['regionByPx', 'regionSquare', 'regionByPct',
+        'sizeByW', 'sizeByH', 'sizeByPct', 'sizeByConfinedWh', 'sizeByWh'],
+      formats: ['jpg'],
+      qualities: ['default', 'bitonal']
+    }
+  },
+  none: {
+    none: {
+      supports: [],
+      formats: [],
+      qualities: []
+    }
+  }
+};
+
+/**
+ * @enum {string}
+ */
+export const Versions = {
+  VERSION1: 'version1',
+  VERSION2: 'version2',
+  VERSION3: 'version3'
+};
+
+const COMPLIANCE_VERSION1 = new RegExp('^https?\:\/\/library\.stanford\.edu\/iiif\/image-api\/(1\.1\/)?compliance\.html#level[0-2]$');
+const COMPLIANCE_VERSION2 = new RegExp('^https?\:\/\/iiif\.io\/api\/image\/2\/level[0-2](\.json)?$');
+const COMPLIANCE_VERSION3 = new RegExp('(^https?\:\/\/iiif\.io\/api\/image\/3\/level[0-2](\.json)?$)|(^level[0-2]$)');
+
+function generateVersion1Options(iiifInfo) {
+  let levelProfile = iiifInfo.getComplianceLevelSupportedFeatures();
+  // Version 1.0 and 1.1 do not require a profile.
+  if (levelProfile === undefined) {
+    levelProfile = IIIF_PROFILE_VALUES.version1.level0;
```

Comment:
Here you should use `Versions.VERSION1` instead of `version1`.

It may be that most minifiers are not aggressive enough to rename `level0`, but it could be subject to renaming as well.  Which suggests that you use an enum for levels as well (or quoted strings everywhere).

Reply:
Done.
Resolved the potential `level0` minifying problem by using quoted strings, not as enum, as they are currently not exported / used anywhere else.